src/Orleans.Core/Runtime/GrainReferenceRuntime.cs

```diff
@@ -18,6 +18,7 @@ internal class GrainReferenceRuntime : IGrainReferenceRuntime
         private readonly ILogger logger;
         private readonly IInternalGrainFactory internalGrainFactory;
         private readonly SerializationManager serializationManager;
+        private readonly IGrainCancellationTokenRuntime cancellationTokenRuntime = new GrainCancellationTokenRuntime();
```

Comment:
you could register a singleton in the container and reduce the number of these runtimes by just using DI

Reply:
BTW, I now saw that you registered the singleton on the silo side. You need to register in the client side, and then you should be able to resolve from this class' constructor